### PR TITLE
new project system

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: GitVersion.yml [hunk 1]
```diff
@@ -2,6 +2,4 @@ assembly-versioning-scheme: Major
 next-version: 6.0
 branches:
   develop:
-    tag: alpha
-  releases?[/-]:
```
Comment (andreasohlund) [inline GitVersion.yml]: Is this on purpose? (in core we have RC release branches since we felt that would be the appropriate default)

Comment (SimonCropp) [inline GitVersion.yml]: gitversion doesnt support that syntax anymore

Comment (andreasohlund) [inline GitVersion.yml]: Got it!

@Particular/nservicebus-maintainers so we need to make this change to core as well once we upgrade to GV 4 

https://github.com/Particular/NServiceBus/blob/develop/GitVersion.yml

@SimonCropp do you know if there is a equivalent setting in v4? (assuming it's not the default)

**Comment by SimonCropp** [comment]
@Particular/container-maintainers this is good to go

File: GitVersion.yml (hunk 1 above)
Comment (SimonCropp) [inline GitVersion.yml]: sorry andreas. i didnt need to know for the changes i am doing so didnt bother looking it up

Comment (bording) [inline GitVersion.yml]: Setting this in V4 is still supported, they just changed the syntax. Each type of branch now has a group name, and and the regex is a property of it.

See https://github.com/Particular/NServiceBus.RabbitMQ/blob/develop/GitVersion.yml for how to do this in V4.

Comment (bording) [inline GitVersion.yml]: See https://gitversion.readthedocs.io/en/latest/configuration/ for more info.

File: src/NServiceBus.StructureMap/NServiceBus.StructureMap.csproj [hunk 2]
```diff
@@ -1,99 +1,35 @@
...
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <DocumentationFile>..\..\binaries\NServiceBus.ObjectBuilder.StructureMap.xml</DocumentationFile>
-    <NoWarn>1591,1573</NoWarn>
+    <DebugSymbols>True</DebugSymbols>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <Prefer32Bit>false</Prefer32Bit>
-    <UseVSHostingProcess>false</UseVSHostingProcess>
+    <DocumentationFile>$(ProjectDir)bin\$(Configuration)\$(AssemblyName).xml</DocumentationFile>
```
Comment (bording) [inline src/NServiceBus.StructureMap/NServiceBus.StructureMap.csproj]: You should be able to replace this with a simpler `<GenerateDocumentationFile>true</GenerateDocumentationFile>` which will name the file after the assembly and put it in the correct location.

File: src/NServiceBus.StructureMap/NServiceBus.StructureMap.csproj [hunk 3]
```diff
@@ -1,99 +1,35 @@
...
-    </PropertyGroup>
-    <Error Condition="!Exists('..\packages\Particular.CodeRules.0.1.1\build\Particular.CodeRules.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Particular.CodeRules.0.1.1\build\Particular.CodeRules.props'))" />
-    <Error Condition="!Exists('..\packages\NuGetPackager.0.6.0\build\NuGetPackager.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\NuGetPackager.0.6.0\build\NuGetPackager.targets'))" />
-    <Error Condition="!Exists('..\packages\Fody.1.29.4\build\dotnet\Fody.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Fody.1.29.4\build\dotnet\Fody.targets'))" />
-    <Error Condition="!Exists('..\packages\GitVersionTask.3.6.3\build\dotnet\GitVersionTask.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\GitVersionTask.3.6.3\build\dotnet\GitVersionTask.targets'))" />
+  <Target Name="BeforeBuildPackages" BeforeTargets="BuildPackages">
```
Comment (bording) [inline src/NServiceBus.StructureMap/NServiceBus.StructureMap.csproj]: You can drop this if you instead set `<OutputPath>..\..\binaries\</OutputPath>`. You do need to tweak the nuspec to add the targetframework path, but this is something that should probably be done anyway to prepate for multi-targeting.

**Pull request merged**